scipy/stats/stats.py

```diff
@@ -4541,6 +4542,131 @@ def ttest_rel(a, b, axis=0, nan_policy='propagate'):
     return Ttest_relResult(t, prob)
 
 
+def ttest_trimmed(a, b, axis=0, equal_var=False, nan_policy='propagate', trimming_percentage=20.0):
+    """
+    Calculate the Yuen T-test on TWO samples of scores, a and b.
+
+    Data sets with low kurtosis (i.e., a distribution flatter than the normal 
+    distribution) should be analyzed with the two-sample trimmed t-test
+    for unequal population variances, also called Yuen t-test.
+    Yuen t-test, also called "20 percent trimmed means test", is an extension
+    of Welch t-test and is allegedly more robust in case of non-normal
+    distributions.
+    Yuen t-test consists of removing the lowest and highest 20 percent
+    of the data and applying Welch t-test on the remaining values.
+
+    Parameters
+    ----------
+    a, b : array_like
+        The arrays must have the same shape.
+    axis : int or None, optional
+        Axis along which to compute test. If None, compute over the whole
+        arrays, `a`, and `b`.
+    nan_policy : {'propagate', 'raise', 'omit'}, optional
+        Defines how to handle when input contains nan. 'propagate' returns nan,
+        'raise' throws an error, 'omit' performs the calculations ignoring nan
+        values. Default is 'propagate'.
+    trimming_percentage : percent of trimming for default it is set to 20% For information about
+        trimming values (see table 2 in paper Yuen 1974)
+    equal_var : bool, optional
+        If True (default), perform a standard independent 2 sample test
+        that assumes equal population variances.
+        If False, perform Welch's t-test, which does not assume equal
+        population variance.
+
+    Returns
+    -------
+    statistic : float or array
+        t-statistic
+    pvalue : float or array
+        two-tailed p-value
+
+    
+    References
+    ----------
+    Karen K. Yuen (1974), The two-sample trimmed t for unequal population
+    variances, Biometrika Volume 61, Number 1, 165-170,
+    DOI: https://doi.org/10.1093/biomet/61.1.165
+
+    Examples
+    --------
+    
+    >>> from scipy import stats    
+
+    >>> a = (56, 128.6, 12, 123.8, 64.34, 78, 763.3)
+    >>> b = (1.1, 2.9, 4.2)
+
+    >>> t, p = stats.ttest_trimmed(a, b, equal_var=False)
+
+    (4.591598691181999, 0.00998909252078421)
+
+    """
+
+    a, b, axis = _chk2_asarray(a, b, axis)
+
+    # check both a and b
+    cna, npa = _contains_nan(a, nan_policy)
+    cnb, npb = _contains_nan(b, nan_policy)
+    contains_nan = cna or cnb
+    if npa == 'omit' or npb == 'omit':
+        nan_policy = 'omit'
+
+    if contains_nan and nan_policy == 'omit':
+        a = ma.masked_invalid(a)
+        b = ma.masked_invalid(b)
+        return mstats_basic.ttest_ind(a, b, axis, equal_var)
+
+    if a.size == 0 or b.size == 0:
+        return Ttest_indResult(np.nan, np.nan)
+
+    n1 = a.shape[axis]
+    n2 = b.shape[axis]
+
+    a.sort()
```

Comment:
this might change a and b outside of the function.

Reply:
yes, it is true, sort will change in place a and b. I need to use sorted